test/unit/test_id_storage.c

```diff
@@ -0,0 +1,62 @@
+#include <r_util.h>
+#include "minunit.h"
+
+
+bool test_r_id_storage_prev_next_eq_0(void) {
```

Comment:
these aren't really testing much apart from the fact that next(prev(x)) == x ... if you don't want to assume any sorting at all on the IDs (and I don't understand the reasons) then this is the wrong structure for this. I don't understand why we can't assume a sorting, given that id_storage have numerical IDs and you *can* sort numbers. sure, id_pool could give numbers in different orders, but that does not matter much, because id_storage maintains an array indexed by the id, so in any case we have a sorting on the ids.

Reply:
the only assumption here, is that prev inverts next. this doesn't mean the same as ids are given out in an ascending/descending or any other specific order. just because the initial pattern provided by r_id_pool is ascending atm, doesn't mean this will stay like that forever.

Reply:
As already explained over telegram, it is essential that if fds are listed in a given order to users then next/prev respect that order, whatever that is.

In other words, if I do `o` and I see 5, 3, 7, 4 and 5 is the current fd, then executing next one time should return 3. Executing next another time should return 7. Then 4 and then 5 again. This is a very important property of this API, otherwise it would be useless.